resources/testharness.js

```diff
@@ -1486,9 +1489,8 @@ policies and contribution forms [3].
         this.index = null;
 
         this.properties = properties;
-        var timeout = properties.timeout ? properties.timeout : settings.test_timeout;
-        if (timeout !== null) {
-            this.timeout_length = timeout * tests.timeout_multiplier;
+        if (settings.test_timeout !== null) {
+            this.timeout_length = settings.test_timeout * tests.timeout_multiplier;
```

Comment:
I suspect that this can be collapsed a bit more, but I'm not sure. @jgraham?

Reply:
You mean like
```js
this.timeout_length = settings.test_timeout !== null ? settings.test_timeout * tests.timeout_multiplier : null;
```
or
```js
this.timeout_length = settings.test_timeout;
if (this.timeout_length !== null) {
  this.timeout_length *= tests.timeout_multiplier;
}
```

Reply:
Updated.

Reply:
I was wondering if more layers couldn't be peeled away, having `timeout_length` on `Test` instances seems like it should be unnecessary. With no test-level timeouts, when the harness times out any async tests that are still running have timed out.

But even after going through all the matches for "timeout" over and over I'm not too confident I understand the setup, so this simplification is fine with me.